src/main/scala/mesosphere/marathon/state/Group.scala

```diff
@@ -217,6 +218,7 @@ object Group {
       group is noAppsAndGroupsWithSameName
       (group.id.isRoot is false) or (group.dependencies is noCyclicDependencies(group))
       group is validPorts
+      group is VolumesModule.validGroup
```

Comment:
I discussed it with @sttts. This should be called by the validation in the root group -- not once per every group. Make sure that validation is really cheap. In Marathon, everything is validated on every change.